pkgs/development/libraries/libfprint-2-tod1-elan/default.nix

```
@@ -0,0 +1,50 @@
+{ stdenvNoCC
+, lib
+, fetchzip
+, libfprint-tod
+, openssl
+, unzip
+}:
+stdenvNoCC.mkDerivation
+{
+  pname = "libfprint-2-tod1-elan";
+  version = "0.0.8";
+
+
+  src = fetchzip {
+    url = "https://download.lenovo.com/pccbbs/mobiles/r1slf01w.zip";
+    hash = "sha256-eY3QsoiBMVuIlT9XY2wI5kc89opgHE7nQa9n71vFpsM=";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    unzip libfprint-2-tod1-elan_$version_Ubuntu22.04.zip
+    patchelf \
+      --set-rpath ${lib.makeLibraryPath [ libfprint-tod openssl ]} \
+      libfprint-2-tod1-elan.so
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm444 libfprint-2-tod1-elan.so -t "$out/lib/libfprint-2/tod-1/"
+
+    runHook postInstall
+  '';
+
+  passthru.driverPath = "/lib/libfprint-2/tod-1";
+
+  meta = with lib; {
+    description = "Elan(04f3:0c4b) driver module for libfprint-2-tod Touch OEM Driver";
+    homepage = "https://support.lenovo.com/us/en/downloads/ds560939-elan-fingerprint-driver-for-ubuntu-2204-thinkpad-e14-gen-4-e15-gen-4";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    license = licenses.unfree;
+    platforms = platforms.linux;
```

```suggestion
    platforms = [ "x86_64-linux" ];
```
The .so will only work on x64.

---

True, i wanted to ask about this but forgot about it again ...
I think this sould be fixed in the other libfprint-tod-drivers aswell.
For example here https://github.com/NixOS/nixpkgs/blob/747e3067461e130d9540c6a284b4e5c33fc0dde7/pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix#L42

---

I guess it doesn't matter all that much. I don't think very many people will try to use these drivers for hardware not mentioned in the package description.

---

I've hit these issues quite a lot while reviewing packages. A lot of the `platforms.<name>` are unintuitive, and the defaults for the builders even more so. There are probably several thousands, if not tens of thousands of packages with incorrect platforms set. Right now, it's more of a best effort thing I think, until someone sits down to revamp it completely. At least it helps hydra know whether it should build for darwin, i386, etc.

---

by the way: whats "the way" to incorporate the changes you guys suggest? just squash everything on the first commit and force push as i did or is there a better way?

---

Yes, that will work (there is no use for fixup commits directly after the init commit). I think there's even some Github functionality to commit suggestions. After that it's just `git pull && git rebase -i origin/master && git push --force`